x/twap/logic_test.go

```diff
@@ -42,76 +43,287 @@ func newExpRecord(accum0, accum1 sdk.Dec) types.TwapRecord {
 	}
 }
 
-func TestRecordWithUpdatedAccumulators(t *testing.T) {
+func (s *TestSuite) TestNewTwapRecord() {
+	// prepare pool before test
+	poolId := s.PrepareBalancerPoolWithCoins(defaultTwoAssetCoins...)
+
+	tests := map[string]struct {
+		poolId        uint64
+		denom0        string
+		denom1        string
+		expectedErr   bool
+		expectedPanic bool
+	}{
```

Comment:
wdyt about changing these expectedErrs from bool to error? That way we can ensure we are receiving the error we are expecting. Panic makes sense as a bool tho imo.

Reply:
sgtm! Changed